pkg/controller/datadogagent/clusterchecksrunner.go

```diff
@@ -374,6 +370,13 @@ func getEnvVarsForClusterChecksRunner(dda *datadoghqv1alpha1.DatadogAgent) []cor
 		})
 	}
 
+	if spec.Agent.Config.DDUrl != nil {
```

Comment:
This conditional checks whether `DDUrl` has been set. As it was set by default previously, the methods would assume it was set. Now we need to check it to make sure we don't cause a panic